app/services/imports/lettings_logs_field_import_service.rb

```diff
@@ -203,5 +205,103 @@ def update_reason(xml_doc)
         @logger.warn("lettings log with old id #{old_id} not found")
       end
     end
+
+    def update_person_details(xml_doc)
+      return if meta_field_value(xml_doc, "form-name").include?("Sales")
+
+      old_id = meta_field_value(xml_doc, "document-id")
+      record = LettingsLog.find_by(old_id:)
+
+      return @logger.warn("lettings log with old id #{old_id} not found") unless record
+      return @logger.info("lettings log #{record.id} has no hhmemb, skipping update") unless record.hhmemb
+
+      if (2..record.hhmemb).all? { |person_index| record.has_any_person_details?(person_index) || record["details_known_#{person_index}"] == 1 }
+        return @logger.info("lettings log #{record.id} has all household member details, skipping update")
+      end
+      if record.hhmemb == 8 || (record.hhmemb + 1..8).none? { |person_index| file_contains_person_details?(xml_doc, person_index) }
+        return @logger.info("lettings log #{record.id} has no additional household member details, skipping update")
+      end
+
+      person_index = 2
+      next_person_index = person_index + 1
+
+      while person_index <= record.hhmemb
+        if next_person_index <= record.hhmemb
+          if record.has_any_person_details?(person_index) || record["details_known_#{person_index}"] == 1
+            @logger.info("lettings log #{record.id} has details for person #{person_index}, skipping update")
```

Comment:
maybe "skipping person" instead of "skipping update" ?

Reply:
Yeah, I like that, thanks!